testing/deployer/sprawl/catalog.go

```diff
@@ -301,11 +471,172 @@ RETRY:
 	return nil
 }
 
+type Resource[V proto.Message] struct {
```

Comment:
I'm not super up to speed on generics... is this a usual pattern?

Reply:
Lifted straight out of core: https://github.com/hashicorp/consul/blob/main/internal/resource/decode.go